### ISSUE TITLE
Consider a different separator than "--" for passing through arguments

### ISSUE THREAD

**Issue by blackdwarf** [dotnet/cli#5061]
Currently with `dotnet run` we have introduced a convention of using `--` to separate the `run` command options from options that should be passed to the command that is being run. 

We should consider and potentially design a better moniker for this than `--`, seeing as how `--` is slightly overloaded. Whatever we figure out, however, needs to be backwards compatible.

/cc @piotrpMSFT @livarcocc @jonsequitur

**Comment by nguerrera** [comment]
There is precedent for -- in many other tools IIRC. If so we should keep it.

**Comment by blackdwarf** [comment]
Actually I found this: http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html and also this StackExchange post: http://unix.stackexchange.com/questions/147143/when-and-how-was-the-double-dash-introduced-as-an-end-of-options-delimiter.

TL;DR: the `--` is a convention that has been added in UNIX back in the 1970s and has been codified in later POSIX standard in 1992. 

It seems that changing this *could* be actually bad, since it would mean we are going against a set up convention.

**Comment by mellinoe** [comment]
How is `--` overloaded? This is the only way I've ever seen it used in any tools. Or do you just mean that in the sense that regular options are `--foo` and this is pretty close? FWIW I definitely don't think we should change this.

**Comment by piotrpMSFT** [comment]
@jonsequitur

**Comment by piotrpMSFT** [comment]
Sounds like there is resounding agreement for us to...
